packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -6,18 +6,31 @@ import { FormResourceService } from '../openmrs-api/form-resource.service';
 import { FormSchemaCompiler } from '@openmrs/ngx-formentry';
 import { LocalStorageService } from '../local-storage/local-storage.service';
 import { FormSchema, Questions } from '../types';
+import { TranslateService } from '@ngx-translate/core';
 
 @Injectable()
 export class FormSchemaService {
   constructor(
     private formsResourceService: FormResourceService,
     private localStorage: LocalStorageService,
     private formSchemaCompiler: FormSchemaCompiler,
+    private translate: TranslateService,
   ) {}
 
   public getFormSchemaByUuid(formUuid: string, cached: boolean = true): Observable<FormSchema> {
     const cachedCompiledSchema = this.getCachedCompiledSchemaByUuid(formUuid);
+    const currentLang = window.i18next?.language.substring(0, 2).toLowerCase() || 'en';
+    this.translate.setDefaultLang(currentLang);
+
     if (cachedCompiledSchema && cached) {
+      if (cachedCompiledSchema?.translations) {
```

Comment:
The caching mechanism probably needs to be re-written. In my view, we should just add the route for the link to the translation resource to the things cached by the offline tools and work with that, but if not, we need to extract the translations and store them in local storage. See my previous comment for how this is to be done.